libbeat/publisher/queue/queue.go

```diff
@@ -50,6 +68,10 @@ type Queue interface {
 
 	Producer(cfg ProducerConfig) Producer
 	Consumer() Consumer
+
+	// Leave this commented out for now so we don't break the interfaces
+	// Should the main Queue implementation itself expose the metrics, or some other component in the interfaces?
+	//Metrics() (Metrics, error)
```

Comment:
I think this is a reasonable place to put it.

Reply:
Alright. Should this PR actually add this to the interface and made some dummy methods for the queues, or should we save that for when we actually implement them?

Reply:
I'd be fine either way, as long as the comments explain the intention. Dummy methods that just return an "unimplemented" error would be fine in this PR so you can start connecting the reporting components, and we can flesh out the queue-specific implementations as more pieces get connected.